src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -211,36 +212,46 @@ public KruizeObject prepareRecommendations(int calCount) {
                 Timestamp interval_start_time = Timestamp.valueOf(Objects.requireNonNull(getInterval_end_time()).toLocalDateTime().minusDays(maxDay));
 
                 // update the KruizeObject to have the results data from the available datasource
-                getResults(mainKruizeExperimentMAP, kruizeObject, experimentName, interval_start_time, dataSource);
+                try {
+                    String errorMsg = getResults(mainKruizeExperimentMAP, kruizeObject, experimentName, interval_start_time, dataSource);
```

Comment:
getResults returning Void 
no need of String

Reply:
Please review to ensure that any caught exceptions are properly reported to the final output.